routes/database.js

```diff
@@ -8,480 +8,382 @@ let my_conf = cfgsrv.get_conf();
-        if(!create_db){
-            await dbsrv.mongo_events().insertOne({'owner': session_user.uid, 'date': new Date().getTime(), 'action': 'database ' + req.params.id + ' declared by ' +  session_user.uid, 'logs': []});
-            res.send({message: 'Database declared'});
+    try {
+        let database = await dbsrv.mongo_databases().findOne({ name: db.name });
+        if (database) {
+            res.status(403).send({ database: null, message: 'Database already exists, please use another name' });
             return;
         }
-        else {
-            let createdb = 'CREATE DATABASE ' + req.params.id + ';\n';
-            try {
-                await querydb(createdb);
-                //await connection.query(createdb);
-            } catch(err) {
-                logger.error('sql error', err);
-                await dbsrv.mongo_events().insertOne({'owner': session_user.uid, 'date': new Date().getTime(), 'action': 'database creation error ' + req.params.id , 'logs': []});
-                res.send({message: 'Creation error: '+err});
-                res.end();
-                return;
-            }
-            //let password = Math.random().toString(36).slice(-10);
-            let password = usrsrv.new_password(10);
-            let createuser = `CREATE USER '${req.params.id}'@'%' IDENTIFIED BY '${password}';\n`;
-            try {
-                await querydb(createuser);
-                //await connection.query(createuser);
-            } catch(err) {
-                logger.error('sql error', err);
-                res.send({message: 'Failed to create user'});
-                res.end();
-                return;
-            }
-            let grant = `GRANT ALL PRIVILEGES ON ${req.params.id}.* TO '${req.params.id}'@'%'\n`;
-            try {
-                await querydb(grant);
-                //await connection.query(grant);
-            } catch(err) {
-                logger.error('sql error', err);
-                res.send({message: 'Failed to grant access to user'});
-                res.end();
-                return;
-            }
-
-            try {
-                await maisrv.send_notif_mail({
-                    'name': 'database_creation',
-                    'destinations': [session_user.email, CONFIG.general.accounts],
-                    'subject': 'Database creation'
-                }, {
-                    '#OWNER#': owner,
-                    '#DBHOST#': CONFIG.mysql.host,
-                    '#DBNAME#': req.params.id,
-                    '#DBUSER#': req.params.id,
-                    '#DBPASSWORD#': password
-                });
-            } catch(error) {
-                logger.error(error);
-            }
 
+        await dbsrv.mongo_databases().insertOne(db);
+        let filter = { name: req.params.id };
+        if (!is_admin) {
+            filter['owner'] = session_user.uid;
+        }
+        await dbsrv.mongo_pending_databases().deleteOne(filter);
```

Comment:
I think you already delete the pending db in the create_db function (which is better, because it's only deleted after creating the db)

The case '!create_db' should only happen when a admin declare a DB, so nothing should need to be deleted.